charts/karpenter/templates/role.yaml

```diff
@@ -48,6 +48,23 @@ rules:
 ---
 apiVersion: rbac.authorization.k8s.io/v1
 kind: Role
+metadata:
```

Comment:
Is it worth adding comments on these roles around why we need them if it isn't immediately clear what we are doing with these things?

Reply:
Why make a separate role?